sw/host/opentitantool/src/command/lc.rs

```diff
@@ -22,12 +22,31 @@ pub struct LcStateReadResult {
     pub lc_state: DifLcCtrlState,
 }
 
-/// Read, decode, and check the LC state is Raw.
-fn check_lc_state_is_raw(jtag: &mut dyn Jtag) -> Result<()> {
+/// Read and decode the current life cycle state, and check whether the requested transition is valid.
+fn check_lc_transition(
+    jtag: &mut dyn Jtag,
+    target: DifLcCtrlState,
+    token: DifLcCtrlToken,
+) -> Result<()> {
     let lc_state =
         DifLcCtrlState::from_redundant_encoding(jtag.read_lc_ctrl_reg(&LcCtrlReg::LcState)?)?;
-    if lc_state != DifLcCtrlState::Raw {
-        bail!("Device must be in Raw LC state. It is in: {}", lc_state);
+    let check = lc_state.check_transition(target);
+    if !check.valid {
+        bail!("Invalid transition from {} to {}.", lc_state, target);
```

Comment:
These `if !cond { bail!(...) }` could just be `ensure!(cond, ...)`.

Also you can take advantage of implicit capture by just write `bail!("Invalid transition from {lc_state} to {target}.");`

Reply:
Changed to `ensure!`